tests/test_transfer_services/test_groupme.py

```diff
@@ -49,66 +69,107 @@ def test_fetch_user_data(mocker, mock_groupme_transfer_service):
     assert oauth2_session_get.call_args.args[1] == 'https://api.groupme.com/v3/users/me'
 
 
-@pytest.mark.parametrize(
-    [
-        'method_name',
-        'json_response',
-        'expected_path',
-        'expected_params',
-        'expected_return_val',
-    ],
-    [
-        (
-            'fetch_blocked_user_vertical',
-            {'response': {'blocks': FAKE_LIST_RESPONSE}},
-            'https://api.groupme.com/v3/blocks',
-            {},
-            FAKE_LIST_RESPONSE,
-        ),
-        (
-            'fetch_chat_bot_vertical',
-            {'response': FAKE_LIST_RESPONSE},
-            'https://api.groupme.com/v3/bots',
-            {},
-            FAKE_LIST_RESPONSE,
-        ),
-        (
-            'fetch_conversation_direct_vertical',
-            {'response': FAKE_LIST_RESPONSE},
-            'https://api.groupme.com/v3/chats',
-            {'per_page': 10},
-            FAKE_LIST_RESPONSE,
-        ),
-        (
-            'fetch_conversation_group_vertical',
-            {'response': FAKE_LIST_RESPONSE},
-            'https://api.groupme.com/v3/groups',
-            {'per_page': 10},
-            FAKE_LIST_RESPONSE,
-        ),
-    ],
-)
-def test_fetch_vertical(
-    method_name,
-    json_response,
-    expected_path,
-    expected_params,
-    expected_return_val,
-    mock_groupme_transfer_service,
-    mocker,
-):
-    full_expected_params = {'user': USER_ID, 'token': TOKEN, **expected_params}
-
+def test_fetch_blocked_users_raises_exception(mock_groupme_transfer_service, mocker):
     mock_groupme_transfer_service._user_id = USER_ID
+    response_object = mocker.MagicMock()
+    response_object.status_code = 200
+    response_object.json.return_value = {'response': {'no_blocks': []}}
+    mock_oauth2_session_get(mocker, response_object)
+    with pytest.raises(ValueError):
+        mock_groupme_transfer_service.fetch_blocked_user_vertical()
+
 
+def test_fetch_blocked_user_vertical(mock_groupme_transfer_service, mocker):
```

Comment:
I'm trying to figure out what this test does test, other than the test harness?  It tests mock_groupme_transfer_service which is test harness mock code, it tests 'dump_and_filter_model_objs" which is test utility code, and it tests 'oauth2_session_get' which is mock code again.

Reply:
I think what I'm missing here is that 'mock_groupme_transfer_service' is NOT a mock.  It's an instance constructed for test purposes with test values, but it's a real instance of GroupMeTransferService and it's the object under test here.   Maybe call it "groupme_transfer_service_test_instance" -- that's long but if you want to have the whole name of the core class here.  I also think it would be fine to call it "groupme_test_instance" or "groupme_transfer_service".  It's pretty common for the instance of a class to be named directly after the class but in snakecase, without having to even call it an instance.